frontend/src/metabase/admin/tasks/components/Logs/hooks.ts

```diff
@@ -0,0 +1,107 @@
+import { useInterval } from "@mantine/hooks";
+import { useState, useEffect, useRef, useCallback } from "react";
+import { useMount } from "react-use";
+import { t } from "ttag";
+import _ from "underscore";
+
+import { UtilApi } from "metabase/services";
+import type { Log } from "metabase-types/api";
+
+import { mergeLogs, hasLog } from "./utils";
+
+export function usePollingLogsQuery(pollingDurationMs: number) {
+  const [loaded, setLoaded] = useState(false);
+  const [error, setError] = useState<any>(null);
+  const [logs, setLogs] = useState<Log[]>([]);
+
+  const isMountedRef = useRef(false);
+  const isFetchingRef = useRef(false);
+
+  const fetchLogs = async () => {
+    if (isFetchingRef.current) {
+      console.warn("skipping logs request as a request is currently in flight");
+      return;
+    }
+
+    try {
+      isFetchingRef.current = true;
+      const newLogs: Log[] = await UtilApi.logs();
+      if (isMountedRef.current) {
+        setLoaded(true);
+        setError(null);
+        setLogs(logs => {
+          const newestLog = _.first(newLogs);
+          const hasFetchedNewLogs = !hasLog(logs, newestLog);
+          if (hasFetchedNewLogs) {
+            return mergeLogs([logs, newLogs.reverse()]);
+          }
+          return logs;
+        });
```

Comment:
not updating the logs if we already have the latest produces really big gains when there's no updates

before optimizations + before skipping if already fetched (~180ms updates):
<img width="1728" alt="Screenshot 2024-02-21 at 2 46 38 PM" src="https://github.com/metabase/metabase/assets/7104357/940e7593-1cb9-4399-a294-0f480d0e1834">

optimizations + before skipping if already fetched (~50ms updates):
<img width="1728" alt="Screenshot 2024-02-21 at 2 43 59 PM" src="https://github.com/metabase/metabase/assets/7104357/c66e2fc3-b754-4046-a740-81c50dcf162f">

optimizations + skipping if already fetched (<1ms updates):
<img width="1728" alt="Screenshot 2024-02-21 at 5 00 49 PM" src="https://github.com/metabase/metabase/assets/7104357/513e7c4d-d892-4ba4-a1c3-03d6645f9d74">
<img width="190" alt="Screenshot 2024-02-21 at 5 00 58 PM" src="https://github.com/metabase/metabase/assets/7104357/62686192-7f10-4c26-92b0-67848f0acdfe">